GatekeeperStudyHall/Assets/Scripts/DiceRoll.cs

```diff
@@ -0,0 +1,203 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+public class DiceRoll : MonoBehaviour
+{
+    bool isHeld = false;
+    bool isSliding = false;
+
+
+    [Header("Shaking")]
+    [SerializeField, Range(0f, 1f)] float shakeInterval = 0.1f;
+    float shakeTimer = 0f;
+
+    // angular velocities the dice is set to while held
+    [SerializeField] float maxAngularVel = 1440f;
+    [SerializeField] float minAngularVel = 900f;
+
+    [SerializeField] float shakeSpeed = 3f;
+    [SerializeField] float shakeRadius = 0.25f;
+
+
+    [Header("Sliding")]
+    [SerializeField] float releaseMultiplier = 12f;
+    [SerializeField, Range(0, 0.1f)] float frictionFactor = 0.002f;
+
+    float slideTimer = 0f;
+    [SerializeField, Range(0f, 1.5f)] float slideDuration = 0.75f;
+
+
+    [Header("Boundaries")]
+    // the dice is restricted between these while sliding
+    [SerializeField] Vector2 topLeftBound;
+    [SerializeField] Vector2 bottomRightBound;
+
+    [SerializeField] GameObject barrier;
+    SpriteRenderer spriteRenderer;
+    Rigidbody2D rb;
+
+    [Space]
+    [SerializeField] Sprite[] sprites; // the 6 dice faces
+
+
+    int roll = -1;
+    public UnityEvent<int> endEvent; // functions to be called after the dice is done rolling
+
+
+    void Start() {
+        if (sprites.Length != 6) {
+            Debug.LogError($"There should be 6 sprites in DiceRoll array (actual: {sprites.Length})");
+        }
+
+        // temporary debug message
+        endEvent.AddListener(x => Debug.Log($"You rolled a {roll}."));
```

Comment:
remove temp debug?

Reply:
I've removed the temporary debug message since we know that the dice roll is being properly calculated.